frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@mui/material';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { getCluster } from '../../lib/cluster';
+import { post } from '../../lib/k8s/apiProxy';
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton, AuthVisible } from '../common';
+
+export function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+  const dispatch = useDispatch();
+
+  function createNewNamespace() {
+    const cluster = getCluster();
+    const newNamespaceData = {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+        name: namespaceName,
+      },
+    };
+    const newNamespaceName = newNamespaceData.metadata.name;
+
+    async function namespaceRequest() {
+      try {
+        const response = await post('/api/v1/namespaces', newNamespaceData, true, {
+          cluster: `${cluster}`,
+        });
+        return response;
+      } catch (error) {
+        console.error('Error creating namespace', error);
+        throw error;
+      }
+    }
+
+    setNamespaceDialogOpen(false);
+    dispatch(
+      clusterAction(() => namespaceRequest(), {
+        startMessage: t('translation|Creating namespace {{  name  }}…', { name: newNamespaceName }),
+        cancelledMessage: t('translation|Cancelled creation of {{  name  }}.', {
+          name: newNamespaceName,
+        }),
+        successMessage: t('translation|Created namespace {{  name  }}.', {
+          name: newNamespaceName,
+        }),
+        errorMessage: t('translation|Error creating namespace {{  name  }}.', {
+          name: newNamespaceName,
+        }),
+        cancelCallback: () => {
+          setNamespaceDialogOpen(true);
+        },
+      })
+    );
+  }
+
+  return (
+    <AuthVisible item={Namespace} authVerb="create">
+      <ActionButton
+        color="primary"
+        description={t('translation|Create')}
+        icon={'mdi:plus-circle'}
+        onClick={() => {
+          setNamespaceDialogOpen(true);
+        }}
+      />
+
+      <Dialog open={namespaceDialogOpen} onClose={() => setNamespaceDialogOpen(false)}>
+        <DialogTitle>{t('translation|Create Namespace')}</DialogTitle>
+        <DialogContent>
+          <Box component="form" style={{ width: '20vw', maxWidth: '20vw' }}>
+            <TextField
```

Comment:
It'd be nice to have some validation for the namespace name, like max characters and allowed characters

Reply:
I tried using some special chars and it doesnt seem to create the namespace with special chars, the length could be long though I tried that also. What would be a good word length for names?

Reply:
According to the docs it should be a valid DNS name, more on that here
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names

Reply:
Fixed!